### ISSUE TITLE
Dynamic method invocation as static issue with grandparents

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#1355]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) v3.0.13 |
| Language level | e.g. PHP 7.0                        |

```
The calls to C1::doX and/or C2::doX (it varies) are caught by the Dynamic methods invocation via '::' inspection

<?php

error_reporting(E_ALL | E_STRICT);

class C1 {
public function doX() {}
}

class C2 extends C1 {
    public function doX() {}
}

class C3 extends C2 {
    public function doX() {}
}

class  C4 extends C3 {
    public function doX() {}
    public function doY() {
        $this->doX();
        parent::doX();
        C1::doX();
        C2::doX();
    }
}

$c4 = new C4();
$c4->doY();
```

Even though PHP interprets it as a static call

```
  19    12        EXT_STMT                                                 
        13        INIT_STATIC_METHOD_CALL                      'C1', 'doX'
        14        EXT_FCALL_BEGIN                                          
        15        DO_FCALL                                      0          
        16        EXT_FCALL_END                                            
```

and PHP 7.0 has cracked down on this (https://www.php.net/manual/en/language.oop5.basic.php)

>The pseudo-variable $this is available when a method is called from within an object context. $this is a reference to the calling object (usually the object to which the method belongs, but possibly another object, if the method is called statically from the context of a secondary object). As of PHP 7.0.0 calling a non-static method statically from an incompatible context results in $this being undefined inside the method. Calling a non-static method statically from an incompatible context has been deprecated as of PHP 5.6.0. As of PHP 7.0.0 calling a non-static method statically has been generally deprecated (even if called from a compatible context). Before PHP 5.6.0 such calls already triggered a strict notice.

I note the phrase "As of PHP 7.0.0 calling a non-static method statically has been generally deprecated (even if called from a compatible context)." and the lack of warnings when executed as evidence of this being an if not completely blessed at least understood to be necessary use case.

**Comment by kalessil** [comment]
Yep, it's a bug.

**Comment by kalessil** [comment]
Fixed!
